keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,969 @@
+pods running on the cluster.
+
+#### Story 5: Restrict egress to well known destinations
+
+As a cluster admin, I want to explicitly limit which workloads can connect to well
+known destinations outside the cluster.
+
+This user story is particularly relevant in hybrid environments where customers
+have highly restricted databases running behind static IPs in their networks
+and want to ensure that only a given set of workloads is allowed to connect to
+the database for PII/privacy reasons. Using ClusterNetworkPolicy, a user can
+write a policy to guarantee that only the selected pods can connect to the database IP.
+
+### Notes/Constraints/Caveats
+
+It is important to note that the controller implementation for cluster-scoped
+policy APIs will not be provided as part of this KEP. Such controllers which
+realize the intent of these APIs will be provided by individual CNI providers,
+as is the case with the NetworkPolicy API.
+
+### Risks and Mitigations
+
+A potential risk of the ClusterNetworkPolicy resource is, when it's stacked on
+top of existing NetworkPolicies in the cluster, some existing allowed traffic
+patterns (which were regulated by those NetworkPolicies) may become blocked by
+ClusterNetworkPolicy `Deny` rules, while some isolated workloads may become
+accessible instead because of ClusterNetworkPolicy `Allow` rules.
+
+Developers could face some difficulties figuring out why the NetworkPolicies
+did not take effect, even if they know to look for ClusterNetworkPolicy rules
+that can potentially override these policies:
+To understand why traffic between a pair of Pods is allowed/denied, a list of
+NetworkPolicy resources in both Pods' Namespace used to be sufficient
+(considering no other CRDs in the cluster tries to alter traffic behavior).
+The same Pods, on the other hand, can appear as an AppliedTo, or an ingress/egress
+peer in any ClusterNetworkPolicy. This makes looking up policies that affect a
+particular Pod more challenging than when there's only NetworkPolicy resources.
+
+In addition, in an extreme case where a ClusterNetworkPolicy `Deny` rule,
+ClusterNetworkPolicy `Allow` rule, NetworkPolicy rule and DefaultNetworkPolicy
+rule applies to an overlapping set of Pods, users will need to refer to the
+precedence model mentioned in the [previous section](precedence-model) to
+determine which rule would take effect. As shown in that section, figuring out
+how stacked policies affect traffic between workloads might not be very straightfoward.
+
+To mitigate this risk and improve UX, a tool which reversely looks up affecting
+policies for a given Pod and prints out relative precedence of those rules
+can be quite useful. The [cyclonus](https://github.com/mattfenwick/cyclonus) project
+for example, could be extended to support ClusterNetworkPolicy and DefaultNetworkPolicy.
+This is an orthogonal effort and will not be addressed by this KEP in particular.
+
```

Comment:
Here a question that came to my mind: we've discussed in past about CNIs providing some useful information about a NetworkPolicy on the status field. @danwinship proposed something here: https://github.com/kubernetes/enhancements/pull/2137

So, by our discussions I can't remember if the status was proposed/designed, but I guess that it is important that we try to predict this on the ClusterNetPol proposal.

In my mind, as a cluster admin, it would be good to have some feedback from the CNI controller at least if and how a NetworkPolicy has been parsed, as an example if it was fully implemented, if it got some parsing issue, etc.

The user stories provided [here](https://github.com/kubernetes/enhancements/pull/2137/files#diff-80527556c28fa329efb6e422217b3434c6b3d28efe9c840ca4f0905138a15aa7R179-R198) IMO are the most important to be taken care of a status field, and maybe even helping to report some misconfiguration as writen in https://github.com/kubernetes/enhancements/pull/2137/files#diff-80527556c28fa329efb6e422217b3434c6b3d28efe9c840ca4f0905138a15aa7R473

Reply:
Just to clarify, as a user I don't expect metrics on the status field (time to apply, if this was applied in all nodes, etc) but a feedback from the Controller if the object was right, if all the selectors could be parsed, if there's some misconfiguration.

Reply:
i was actually hoping to see @danwinship proposal evolve and inherit the status part for CNPs :D it will be useful for users i agree

Reply:
updated KEP to include status